homeassistant/components/inels/config_flow.py

```diff
@@ -0,0 +1,273 @@
+"""Config flow for iNELS."""
+from __future__ import annotations
+
+from collections import OrderedDict
+from typing import Any
+
+from inelsmqtt import InelsMqtt
+from inelsmqtt.const import MQTT_TRANSPORT
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.hassio.discovery import HassioServiceInfo
+from homeassistant.const import (
+    CONF_DISCOVERY,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, TITLE
+
+CONNECTION_TIMEOUT = 5
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle of iNELS config flow."""
+
+    VERSION = 1
+
+    _hassio_discovery: dict[str, Any] | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> InelsOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return InelsOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow by user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        return await self.async_step_setup()
+
+    async def async_step_setup(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Configure the setup."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            test_connect = await self.hass.async_add_executor_job(
+                try_connection,
+                self.hass,
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+                user_input.get(CONF_USERNAME),
+                user_input.get(CONF_PASSWORD),
+                user_input.get(MQTT_TRANSPORT),
+            )
+
+            if test_connect is None:
+                user_input[CONF_DISCOVERY] = True
+                return self.async_create_entry(
+                    title=TITLE,
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_PORT: user_input[CONF_PORT],
+                        CONF_USERNAME: user_input.get(CONF_USERNAME),
+                        CONF_PASSWORD: user_input.get(CONF_PASSWORD),
+                        MQTT_TRANSPORT: user_input.get(MQTT_TRANSPORT),
```

Comment:
You should be consistent between .get and [] just stylistically speaking. I don't know if this is something maintainers would actually have a problem with though. You do it  above on line 62 as well.